### DISCUSSION TITLE
Contributing Projects

### DISCUSSION THREAD

**Discussion by juergen-albert** [eclipse-emfcloud/emfcloud#230] [All other]
Hi,

we have a bunch of EMF related Projects, we are looking to find a home for in the Eclipse Foundation. Before we start a new Project, it makes sense to look fore something existing, as I believe we fit right to the given Projects purpose. Especially as we already use e.g. emfjson and expand it. 

So what do we have and want to contribute:

# GeckoEMF - https://github.com/geckoprojects-org/org.gecko.emf

* Dynamic EMF for a pure OSGi environment, which does not require any Extensionregistry

#  GeckoEMFUtil - https://github.com/geckoprojects-org/org.gecko.emf.utils

* JakartaRS Support for EMF Objects and lots of ContentTypes (e.g. json, yml, bson, XMI, XML, Binary, properties etc.)
* various exporters for e.g. ODS, Excel, r-Data, CSV,)

# GeckoPersistence - https://github.com/geckoprojects-org/org.gecko.emf.persistence

* EMF based Persistence for various Databases
    * In parts based on the old mongoEMF Project of Bryan Hunt
    * MongoDB
    * JPA and everything it supports (currently in development)

All the mentioned Projects are used in other Eclipse Projects as well (Eclipse Sensinact and Eclipse Daanse). 

We are still actively developing all of these and will do so in the foreseeable future. All of the projects are actively maintained and used in various projects for years now.

How do you feel about this?

**Comment by planger** [comment]
Hi Jürgen,

First off, a huge thanks for thinking of EMF Cloud as a potential home for your projects! It's always exciting to see new contributions and the willingness to collaborate within the Eclipse Foundation community.

However, after looking into the components a bit, I feel that they, while undoubtedly valuable, might be better suited as their own project within the Eclipse Foundation. Aside from the dependency on emfjson, it seems there's a relatively loose connection to the core goals of EMF Cloud.

Grouping these diverse components under EMF Cloud would mean aligning with the same committership, project management, responsibilities, and community support / communications. Given the unique nature of your projects and the existing components within EMF Cloud, this alignment might not be the perfect fit.

That said, I'm more than open to supporting you in establishing your components as separate projects within the Eclipse Foundation. This approach could provide the focused attention and specific community engagement that your projects deserve. See the [Eclipse Project Handbook](https://www.eclipse.org/projects/handbook/#starting) for starting a new Eclipse project for more information. Feel free to reach out if you're interested in exploring this path further.

Looking forward to potentially collaborating in the future!

Best wishes,

Philip
